iceoryx_examples/callbacks/README.md

```diff
@@ -184,6 +194,113 @@ Afterward, we reset both caches to start fresh again.
     }
 ```
 
+### ice_callbacks_listener_as_class_member.cpp
+
+This example is identical to the [ice_callbacks_subscriber.cpp](#ice_callbacks_subscriber.cpp) 
+one except that we left out the cyclic heartbeat trigger. The key difference is that 
+the listener is now a class member and in every callback we would like to change 
+some member variables. To do this we require an additional pointer to the object 
+since the listener requires c function references which do not allow the usage
+of lambdas with capturing. Here we can use the userType feature which allows us 
+to provide the this pointer as additional argument to the callback.
+
+The main function is now pretty short, we instantiate our object of type `CounterClass`
+and call `waitForControlC` which uses the `shutdownSemaphore` like in the 
```

Comment:
at first I thought you accidentally added a `C`. How about calling this just `wait` ot `waitForShutdown`